src/lib/components/div/Div.js

```diff
@@ -0,0 +1,112 @@
+//@flow
+import styled from 'styled-components';
+
+const spacing = {
+  sp1: '0.71rem',
+  sp2: '0.85rem',
+  sp3: '1rem',
+  sp4: '1.14rem',
+  sp5: '1.43rem',
+  sp6: '1.57rem', // no use case for the moment
+  sp7: '1.71rem', // no use case for the moment
+};
+
+type Props = {|
+  m?: number,
+  mt?: number,
+  mr?: number,
+  mb?: number,
+  ml?: number,
+  mx?: number,
+  my?: number,
+  p?: number,
+  pt?: number,
+  pr?: number,
+  pb?: number,
+  pl?: number,
+  px?: number,
+  py?: number,
+|};
+
+type PropsName =
+  | 'm'
+  | 'mt'
+  | 'mr'
+  | 'mb'
+  | 'ml'
+  | 'mx'
+  | 'my'
+  | 'p'
+  | 'pt'
+  | 'pr'
+  | 'pb'
+  | 'pl'
+  | 'px'
+  | 'py';
+
+type CSSField =
+  | 'margin'
+  | 'margin-top'
+  | 'margin-right'
+  | 'margin-bottom'
+  | 'margin-left'
+  | 'padding'
+  | 'padding-top'
+  | 'padding-right'
+  | 'padding-bottom'
+  | 'padding-left';
+
+const createSpacingFn = (
+  propsName: PropsName,
+  spacingFields: Array<CSSField>,
+) => (props: Props) => {
+  if (!props[propsName]) {
+    return null;
+  }
+  let value: number = props[propsName];
+
+  if (value < 1 || value > 7) {
+    console.warn(`'${propsName}' props should be a number between 1 and 7`);
+    return null;
+  }
+
+  const result = spacingFields
+    .map((param: CSSField) => `${param}: ${spacing[`sp${value}`]}`)
+    .join(';');
+  return result;
+};
+
+const m = createSpacingFn('m', ['margin']);
+const mt = createSpacingFn('mt', ['margin-top']);
+const mr = createSpacingFn('mr', ['margin-right']);
+const mb = createSpacingFn('mb', ['margin-bottom']);
+const ml = createSpacingFn('ml', ['margin-left']);
+const mx = createSpacingFn('mx', ['margin-top', 'margin-bottom']);
+const my = createSpacingFn('my', ['margin-left', 'margin-right']);
```

Comment:
I think x and y axis are reversed here. In the common spirit x should stand for the horizontal axis I believe. (Tailwind also implemented the x axis against left/right properties : https://tailwindcss.com/docs/padding#add-horizontal-padding)

Reply:
Indeed that was a mistake